package-res/resources/web/vizapi/DataTable.js

```diff
@@ -15,1094 +15,1220 @@
 *
 */
 
-/*
-pentaho.DataTable
-pentaho.DataView
+pentaho = typeof pentaho == "undefined" ? {} : pentaho;
 
-Client-side non-visual data tables.
+/**
+ * Pentaho JavaScript Visualization Data APIs.
+ *
+ * This module defines interfaces and classes
+ * used by visualizations for data representation and manipulatation.
+ *
+ * @module common-ui.vizapi.data
+ * @main common-ui.vizapi.data
+ */
+
+/**
+ * An _AbstractDataTable_ represents a set of tabular data.
+ *
+ * @class AbstractDataTable
+ * @constructor
+ */
+ pentaho.AbstractDataTable = function AbstractDataTable() {};
+
+/**
+ * Gets a column object, given its index.
+ *
+ * @method _getColumn
+ * @param {Number} colIdx The column index (zero-based).
+ * @return {Object} The column object.
+ * @protected
+ */
+
+/**
+ * Gets a row object, given its index.
+ *
+ * @method _getRow
+ * @param {Number} rowIdx The row index (zero-based).
+ * @return {Object} The row object.
+ * @protected
+ */
+
+/**
+ * Gets a cell, given its row and column index.
+ *
+ * @method _getCell
+ * @param {Number} rowIdx The cell's row index (zero-based).
+ * @param {Number} colIdx The cell's column index (zero-based).
+ * @return {Null|Object} The cell.
+ * @protected
+ */
+
+/**
+ * Gets the number of columns.
+ *
+ * @method getNumberOfColumns
+ * @return {Number} The number of columns.
+ */
+
+/**
+ * Gets the number of rows.
+ *
+ * @method getNumberOfRows
+ * @return {Number} The number of rows.
+ */
+
+// TODO: Not publicizing the add method yet, as it is not unit-tested,
```

Comment:
This is expected to be fixed some time in the future. Probably only in a subsequent release of the VizAPI. There's no backlog issue for it. The method was added and documented for this release, to satisfy internal uses (see createTrend), and is more or less conformant with the corresponding method of the [Google DataTable](https://developers.google.com/chart/interactive/docs/reference#DataTable_addColumn).

The comment clarifies that the doclet below purposely has a single starting "*" — to not publicize the method. It helps prevent someone, thinking it is a bug, making it public (and even if there were a backlog item for this, it wouldn't substitute the need for an inline call for attention).

Comments that don't have some starting "TAG:" are used more to describe the workings of the code, not as advertencies, and have a permanent character.

I strive to make it crystal clear what the programmer's intentions were, at the time he wrote the code. I find this value more important that any NO TODOs rule (or prejudice?). Don't you think?

Should I make the comment clearer?
I probably should use a "NOTE: " tag instead...?

Thanks!